### PR TITLE
Simplify and shorten process time tests

### PR DESCRIPTION
**Pull request by rwy0717**
- Disable TestThreadCpuTime by default. It's run time is too long. Fixes
  issue #181, where we are timing out on Travis-CI.
- Write simpler, shorter tests for monotonic CPU time.
- Make better use of the gtest assertions.
- Fix broken application CPU time test.
- Format in the WebKit style.

Signed-off-by: Robert Young <rwy0717@gmail.com>

### CODE REVIEW

**Comment by mgaudet** [comment]
(Not asking for this commit, but) Maybe we should make sure formats end up in their own commits. 

Would help if we wanted to deploy a system like [`hyper-blame`](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/git-hyper-blame.html), but also makes review easier.

**Comment by charliegracie** [comment]
@rwy0717 I am not 100% sure we are going with WebKit style at this point. We may want to not format this file right now just in case another style gets selected.  Could you prepare a commit that does not change the format?

**Comment by rwy0717** [comment]
Sure can.

**Comment by charliegracie** [comment]
@rwy0717 tall 3 linux builds failed due to CPU Time test.  OSX has not run yet.

**Comment by rwy0717** [comment]
I'm guessing GCC nop'd my cpu burn function, haha. Can anyone provide an implementation that will not optimize out?

This was mine, which worked at `-O0`:
```
void
userTimeCpuBurn()
{
        volatile double x = 1.1;
        for (size_t i = 0; i < 500; i++) {
                x = pow(x, 1.1);
        }
}
```

The previous burn function worked but, it's implementation relied on placing a very big array on the stack. I would rather find a way to explicitly tell the compiler to not optimize inline. Here's the old one:
```
void
matrixSquare(void)
{
	/*
	 * Declare and initialize the matrix. memset works here since this is a true 2D
	 * array defined in the same scope as the memset.
	 */
	uintptr_t matrix[MATRIX_SIZE][MATRIX_SIZE];
	uintptr_t i = 0;
	uintptr_t j = 0;

	memset(matrix, 0, sizeof(matrix));

	for (j = 0; j < MATRIX_SIZE; j++) {
		matrix[i][j] = (uintptr_t)pow((double)100, 2.0);
	}
}
```

**Comment by charliegracie** [comment]
Would you be able to use j9thread_sleep() to test it?  

If you really need a burn function you can look at the tight loop in the spin lock code which uses the atomic nop or yield function which should not be optimized out.

**Comment by rwy0717** [comment]
OK thanks. We are checking for increasing cpu time, so sleep isn't a good choice. I'll look at the spin lock.

**Comment by mgaudet** [comment]
Not really portable, but you're looking for something in the neighborhood of `asm volatile("nop")`

**Comment by rwy0717** [comment]
It is working on my mac os machine with `-O3`. `Apple LLVM version 8.0.0 (clang-800.0.38)`.

**Comment by rwy0717** [comment]
Made the following changes:
* Update the burn function to use the atomic nop.
* Use a test fixture to enable cpu tracking and set the thread category.
* Use asserts to short circuit the test loop. Should fail less verbosely :3

**Comment by rwy0717** [comment]
Squashed, formatted, signed off. Will launch an internal build and report back with success/fail.

**Comment by rwy0717** [comment]
The new tests fail when CPU time is constant, or zero. The old tests succeed.  Is it correct to expect that CPU time increases, or is nonzero?

I've pushed the old tests to this branch, but modified to fail on constant CPU time.

Is there the potential for a missing initialization call in the thread library?

**Comment by rwy0717** [comment]
The bottom line is, it's impossible to generate a load taking measurable time when you don't trust the clock. I've changed the tests to check for non-decreasing time, instead of increasing time.

File: fvtest/threadextendedtest/processTimeTest.cpp [hunk 1]
```diff
@@ -81,115 +45,126 @@ systemTimeCPUBurn(void)
...
-	 * If we increase the workload done (running the same function multiple times),
-	 * CPU time recorded above should increase after each run
-	 */
-	for (i = 0; i < arrayLength; i++) {
-		systemTimeCPUBurn();
-		retValOfGetProcessTimes = (omrthread_get_process_times(processTimeArray++) || retValOfGetProcessTimes);
-		EXPECT_TRUE(0 == retValOfGetProcessTimes);
+TEST(ThreadCpuTime, userCpuTimeIncreasesMonotonically)
+{
+	omrthread_process_time_t cpuTime, prevCpuTime;
```
Comment (charliegracie) [inline fvtest/threadextendedtest/processTimeTest.cpp]: To make this code more clear would you place each declaration on a separate line?  There are other cases below that should also be fixed.

File: fvtest/threadextendedtest/processTimeTest.cpp [hunk 2]
```diff
@@ -320,13 +297,16 @@ TEST(ThreadExtendedTest, TestThreadCpuTime)
 	/* Enable CPU monitoring as it is disabled by default */
 	omrthread_lib_enable_cpu_monitor(omrthread_self());
 
-	/* Create the SupportThreadInfo structure for the required number of threads and intialize it */
-	thrInfo = (SupportThreadInfo *)omrmem_allocate_memory(sizeof(SupportThreadInfo), OMRMEM_CATEGORY_THREADS);
+	/* Create the SupportThreadInfo structure for the required number of threads
+* and intialize it */
```
Comment (charliegracie) [inline fvtest/threadextendedtest/processTimeTest.cpp]: Can you fix the spacing/tabbing?

File: fvtest/threadextendedtest/processTimeTest.cpp [hunk 3]
```diff
@@ -386,35 +360,8 @@ TEST(ThreadExtendedTest, TestThreadCpuTime)
 	/* Application threads' cpu usage should increase */
 	ASSERT_TRUE(cpuUsageAft.applicationCpuTime > cpuUsageBef.applicationCpuTime);
 	/*
-	 * system cpu threads stopped after we checked the first time, but their time
-	 * should either be the same or (slightly) higher
-	 */
+* system cpu threads stopped after we checked the first time, but their time
```
Comment (charliegracie) [inline fvtest/threadextendedtest/processTimeTest.cpp]: Tabbing?

File: fvtest/threadextendedtest/processTimeTest.cpp [hunk 4]
```diff
@@ -301,14 +278,14 @@ appThread(void *arg)
...
-#define NUM_SYS_THREAD	10
+#define NUM_APP_THREAD 50
+#define NUM_SYS_THREAD 10
 
 /**
  * Compare the values of omrthread_get_cpu_time and omrthread_get_self_cpu_time
  * and the monotonicity of the omrthread_get_cpu_time values
  */
-TEST(ThreadExtendedTest, TestThreadCpuTime)
+TEST(ThreadExtendedTest, DISABLED_TestThreadCpuTime)
```
Comment (charliegracie) [inline fvtest/threadextendedtest/processTimeTest.cpp]: We are going to leave this test disabled for now?

Comment (rwy0717) [inline fvtest/threadextendedtest/processTimeTest.cpp]: Yes. Disabling until it's run time is reduced.

There are two new CpuTimeTest unit tests for coverage on `omrthread_get_jvm_cpu_usage_info`.

Both TestThreadCpuTime and CpuTimeTest check for monotonic cpu time growth across the entire process. The difference is TestThreadCpuTime brings up multiple threads to burn the CPU, the CpuTimeTest is single-threaded.

File: fvtest/threadextendedtest/processTimeTest.cpp (hunk 2 above)
Comment (rwy0717) [inline fvtest/threadextendedtest/processTimeTest.cpp]: d'oh.. fixed :)

File: fvtest/threadextendedtest/processTimeTest.cpp (hunk 1 above)
Comment (rwy0717) [inline fvtest/threadextendedtest/processTimeTest.cpp]: fixed.

**Pull request merged**